File: Data Structures & Algorithms/binary-tree-maximum-path-sum/submission-2.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int res;
    int calc(TreeNode* root){
        if(!root)
            return 0;
        int lv = max(calc(root->left), 0);
        int rv = max(calc(root->right), 0);
        res=max(res,root->val+lv+rv);
        return root->val+max(lv,rv);
    }
    int maxPathSum(TreeNode* root) {
        res=INT_MIN;
        calc(root);
        return res;
    }
};

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def count(self, root, mtl):
        if not root: return 0
        c = 0
        if root.val>=mtl: c = 1
        mtlc = max(mtl, root.val)
        return c + self.count(root.left, mtlc) + self.count(root.right, mtlc)

    def goodNodes(self, root: TreeNode) -> int:
        return self.count(root, -101) 